core/lib/src/fairing/ad_hoc.rs

```diff
@@ -146,9 +145,10 @@ impl Fairing for AdHoc {
     fn on_attach(&self, rocket: Rocket) -> Result<Rocket, Rocket> {
         if let AdHocKind::Attach(ref mutex) = self.kind {
             let mut option = mutex.lock().expect("AdHoc::Attach lock");
-            option.take()
```

Comment:
The extra let binding should be unnecessary -- just replace the line `.call_box((rocket,))` with `(rocket)`.

Reply:
It is unnecessary, however it looks like ugly when having a function call in a multi-line chain like this.